stor/utils.py

```diff
@@ -312,6 +313,52 @@ def is_writeable(path, swift_retry_options=None):
     return answer
 
 
+def validate_file_path(dest, _warn=False):
+    """Raises ValueError if dest is an ambiguous or invalid path.
+
+    Args:
+        dest (Path): path to validate
+        _warn (bool): DO NOT USE: if True, only generate warning (will be removed in a future
+            version)
+    Raises:
+        ValueError: if path is ambiguous and is not a filesystem path (see below)
+    Note:
+        if an ambiguous filesystem path and ``always_raise_on_ambiguous_path`` is true, will raise
+        a ValueError as well, otherwise it warns.
+
+    In OBS-land, we cannot differentiate between files and directories, except by convention.
+    Conventionally, directories are either prefixes or zero-sized objects ending with a trailing
+    slash. With POSIX copies, it's conventional to do something like::
+        cp source/myfile.txt dest
+
+    In which case the system checks whether dest is a directory and, if it is, dumps to
+    ``dest/myfile.txt`` or otherwise dumps to ``dest``.
+
+    We can't do that kind of validation and don't want the outcome of a command to depend upon
+    whether there's a file with the same prefix or not. Thus, we raise an error if you try to use
+    `stor.copy()`. (to manipulate files without extensions, you can always use `stor.copytree()`.
+    """
+    import stor.settings
+
+    # we use a try/except so that it's easier for the reader to tell why the exception was
+    # generated.
+    try:
+        if not dest.name:
```

Comment:
is it slightly less explicit to rely on `dest.name` to verify no trailing slash rather than doing directly?